app/models/akka/RequesterBehavior.scala

```diff
@@ -181,17 +201,25 @@ trait RequesterBehavior { me: AbstractDSLinkActor with Meter =>
   private def cleanupStoredTargets(responses: Seq[DSAResponse]) = {
 
     def cleanupSids(rows: Seq[DSAVal]) = try {
-      rows collect extractSid foreach targetsBySid.remove
+        for (r <- rows)
+          persist(RemoveTargetBySid(extractSid(r))) { event =>
+            log.debug("{}: removing by SID persisted {}", ownId, event)
+            targetsBySid.remove(event.sid)
+          }
```

Comment:
Is it possible to implement somekind of batch event here?

Reply:
It is a good question in general, how deep we should dig into the structures when persist an event.
It is always balance between an additional code and how often we write to the journal.
I tried to figure out the best design in the situation above. I supposed it is OK in case SIDs set is not so big...

Anyway I've redesigned it per your comment. Probably it is a better choice to persist it as a single event.